### PR TITLE
Added DeploymentStatus API specs and examples

### PR DESCRIPTION
**Pull request by shubDhond**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [X] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific language SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [X] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [X] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [X] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that label "WaitForARMFeedback" will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by shubDhond** [comment]
Hi @weidongxu-microsoft, looks like some checks are failing with errors unrelated to my changes but remaining checks are passing. Could you please review when you get a chance?

File: specification/web/resource-manager/Microsoft.Web/stable/2022-03-01/WebApps.json
Comment (raosuhas) [inline specification/web/resource-manager/Microsoft.Web/stable/2022-03-01/WebApps.json:2924]: >operationId [](http://example.com/codeflow?start=127&length=11)

Can you please rename to something like deploymentstatusId

**Comment by raosuhas** [comment]
>           "WebApps"

Seems like the new api is already defined here ?

---
Refers to: specification/web/resource-manager/Microsoft.Web/stable/2022-03-01/WebApps.json:2980 in 144bab0. [](commit_id = 144bab0d326cfbf968388dd3b06636054e8006df, deletion_comment = False)

File: specification/web/resource-manager/Microsoft.Web/stable/2022-03-01/WebApps.json
Comment (raosuhas) [inline specification/web/resource-manager/Microsoft.Web/stable/2022-03-01/WebApps.json:11838]: >operationId [](http://example.com/codeflow?start=140&length=11)

same here please rename to reflect resourceType

**Comment by raosuhas** [comment]
Changes look fine except some minor comments from ARM API review perspective. Please get the breaking change approval and we can signoff

**Review by raosuhas — APPROVED** [review]
:shipit:

File: specification/web/resource-manager/Microsoft.Web/stable/2022-03-01/WebApps.json
Comment (shubDhond) [reply specification/web/resource-manager/Microsoft.Web/stable/2022-03-01/WebApps.json]: Fixed

File: specification/web/resource-manager/Microsoft.Web/stable/2022-03-01/WebApps.json
Comment (shubDhond) [reply specification/web/resource-manager/Microsoft.Web/stable/2022-03-01/WebApps.json]: Fixed

**Comment by weidongxu-microsoft** [comment]
@shubDhond

I removed the "breaking changes" label. Will see if it reappear again later.

In recent months, swagger tool requires all API having example. Given the number of missing examples, you might had to suppress this check for webapps.json in readme.

Loop in @raych1 for guide on ModelValidation errors.

**Comment by raosuhas** [comment]
> @shubDhond
> 
> I removed the "breaking changes" label. Will see if it reappear again later.
> 
> In recent months, swagger tool requires all API having example. Given the number of missing examples, you might had to suppress this check for webapps.json in readme.md (or add 300+ examples?).
> 
> Loop in @raych1 for guide on ModelValidation errors.



> [weidongxu-microsoft](/weidongxu-microsoft)
@weidongxu-microsoft  I had added the breakingchange required label since there was a new route added in an existing api version. Doesnt this need a breaking change approval ?

**Comment by weidongxu-microsoft** [comment]
@raosuhas
The target branch of the PR is a feature branch [Azure:Web-ANT97-2022-03-01](https://github.com/Azure/azure-rest-api-specs/tree/Web-ANT97-2022-03-01) (there is no 2022-03-01 on main or public, hence there is no breaking changes for adding APIs).

Web requires all PR to the feature branch get proper reviewed (include ARM and SDK), hence we are here.

**Review by weidongxu-microsoft — APPROVED** [review]
LGTM

You need to suppress ModelValidation CI, before I can merge.

E.g. https://github.com/Azure/azure-rest-api-specs/pull/18670/files

File: specification/web/resource-manager/Microsoft.Web/stable/2022-03-01/WebApps.json
Comment (weidongxu-microsoft) [inline specification/web/resource-manager/Microsoft.Web/stable/2022-03-01/WebApps.json:22574]: nit, extra dot

File: specification/web/resource-manager/Microsoft.Web/stable/2022-03-01/WebApps.json
Comment (weidongxu-microsoft) [inline specification/web/resource-manager/Microsoft.Web/stable/2022-03-01/WebApps.json:22579]: should it be an enum (which include succeeded, failed, etc.) with model-as-string=false?

**Comment by weidongxu-microsoft** [comment]
@shubDhond

I should be able to merge, after you check the comments and decide whether to change.

**Comment by edwin-msft** [comment]
@weidongxu-microsoft  shubDhond is no longer at Microsoft. Please merge to feature branch [Azure:Web-ANT97-2022-03-01](https://github.com/Azure/azure-rest-api-specs/tree/Web-ANT97-2022-03-01)

I will follow up with fixes to the extra dot(https://github.com/Azure/azure-rest-api-specs/pull/18810#discussion_r865516133),

and to change deployment build status to an enum (https://github.com/Azure/azure-rest-api-specs/pull/18810#discussion_r865516288)

(I've made your suggested changes but don't have permissions to shubDhond's branch
image
)

**Comment by weidongxu-microsoft** [comment]
@edwin-msft

The major issue is the 300+ error in ModelValidation which you probably had to suppress. If you do not fix it in this PR, it just breaks Navy's big one.
See https://github.com/Azure/azure-rest-api-specs/pull/18810#issuecomment-1114106490

You can create a new branch on your forked repo, using this branch. And use that to make a new pull request. I will carry the "ARMSignedOff" there.

Alternatively, you can talked this with Navy and promise him that you are making a follow-up PR to fix these minor swagger issues, as well as this large ModelValidation issue?

**Comment by weidongxu-microsoft** [comment]
replaced by https://github.com/Azure/azure-rest-api-specs/pull/19259

**Pull request closed without merging**
